src/components/patch-details/patch-details.test.tsx

```diff
@@ -99,45 +120,44 @@ describe("Patch Details", () => {
   });
 
   test("it displays the patch, housing officer, and area manager", async () => {
-    render(<PatchDetails asset={assetWithPatches} />);
+    if (!assetWithPatches.patches) throw new Error("No patches");
+    render(
+      <PatchDetails
+        assetPk={assetWithPatches.id}
+        assetPatch={assetWithPatches.patches[0]}
+      />,
+    );
+    await waitFor(async () => {
+      screen.getByTestId("patch-name");
+    });
 
     const patchNameField = screen.getByTestId("patch-name");
     const officerNameField = screen.getByTestId("officer-name");
     const areaManagerNameField = screen.getByTestId("area-manager-name");
 
-    await waitFor(async () => {
-      expect(patchNameField).toHaveTextContent(mockAssetPatch.name);
-      expect(officerNameField).toHaveTextContent(
-        mockAssetPatch.responsibleEntities[0].name,
-      );
-      expect(areaManagerNameField).toHaveTextContent(
-        mockAssetArea.responsibleEntities[0].name,
-      );
-    });
+    expect(patchNameField).toHaveTextContent(mockAssetPatch.name);
+    expect(officerNameField).toHaveTextContent(
+      mockAssetPatch.responsibleEntities[0].name,
+    );
+    expect(areaManagerNameField).toHaveTextContent(
+      mockAssetArea.responsibleEntities[0].name,
+    );
   });
 
   test("it sets a cookie with the asset ID when the edit patches button is clicked", async () => {
     // This is used to redirect the user back to the asset page after editing patches
-    render(<PatchDetails asset={assetWithPatches} />);
-
-    const editPatchesButton = screen.getByTestId("all-patches-and-areas-button");
+    if (!assetWithPatches.patches) throw new Error("No patches");
+    render(
+      <PatchDetails
+        assetPk={assetWithPatches.id}
+        assetPatch={assetWithPatches.patches[0]}
+      />,
+    );
 
     await waitFor(async () => {
+      const editPatchesButton = screen.getByTestId("all-patches-and-areas-button");
       editPatchesButton.click();
       expect(document.cookie).toContain(`fromAssetId=${assetWithPatches.id}`);
     });
   });
-
-  test("it displays a message when there are no patches", async () => {
```

Comment:
Maybe we should keep this test and check that it displays no patches if the property is not assigned to a patch

Reply:
Added test in 416b8e8b41ec6e372081d43a291345481ddd2c10